vic/drivers/shared_all/src/initialize_soil.c

```diff
@@ -38,28 +38,40 @@ initialize_soil(cell_data_struct **cell,
     extern option_struct options;
 
     size_t               veg, band, lindex, frost_area;
-    double               tmp_moist[MAX_LAYERS];
-    double               tmp_runoff;
 
     for (veg = 0; veg <= veg_num; veg++) {
         for (band = 0; band < options.SNOW_BAND; band++) {
-            cell[veg][band].baseflow = 0;
-            cell[veg][band].runoff = 0;
+            // Prognostic states
+            cell[veg][band].aero_resist[0] = 0.0;
+            cell[veg][band].aero_resist[1] = 0.0;
+            cell[veg][band].CLitter = 0.0;
+            cell[veg][band].CInter = 0.0;
+            cell[veg][band].CSlow = 0.0;
             for (lindex = 0; lindex < options.Nlayer; lindex++) {
-                cell[veg][band].layer[lindex].evap = 0;
-                cell[veg][band].layer[lindex].moist = 0;
-                tmp_moist[lindex] = cell[veg][band].layer[lindex].moist;
+                cell[veg][band].layer[lindex].Cs = 0.0;
+                cell[veg][band].layer[lindex].T = 0.0;
+                cell[veg][band].layer[lindex].bare_evap_frac = 0.0;
+                cell[veg][band].layer[lindex].evap = 0.0;
                 for (frost_area = 0; frost_area < options.Nfrost;
                      frost_area++) {
-                    cell[veg][band].layer[lindex].ice[frost_area] = 0;
+                    cell[veg][band].layer[lindex].ice[frost_area] = 0.0;
                 }
+                cell[veg][band].layer[lindex].kappa = 0.0;
+                cell[veg][band].layer[lindex].moist = 0.0;
+                cell[veg][band].layer[lindex].phi = 0.0;
             }
-            compute_runoff_and_asat(soil_con, tmp_moist, 0,
-                                    &(cell[veg][band].asat), &tmp_runoff);
-            wrap_compute_zwt(soil_con, &(cell[veg][band]));
```

Comment:
Actually did you mean you think we should initialize `cell[veg][band].asat`? This part is done in `compute_derived_state_vars`